cardano-node/src/Cardano/Node/Run.hs

```diff
@@ -363,15 +373,15 @@ canonDbPath npm@NodeCLI{databaseFile, nodeMode} = do
   canonicalizePath =<< makeAbsolute dbFp
 
 createDiffusionArguments
-  :: [AddrInfo]  -- TODO: use descrptive type for existing socket vs addr
-  -> FilePath    -- TODO: use descrptive type for existing socket vs path
+  :: Either [Socket] [AddrInfo]
```

Comment:
could you add haddocks to describe what re these arguments (we know, but it might make a difference for people not familiar with the networking bits).